pkg/common/namespace_selection.go

```diff
@@ -5,44 +5,100 @@ package common
 
 import (
 	"context"
+	"fmt"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	policyv1 "open-cluster-management.io/config-policy-controller/api/v1"
 )
 
-// GetSelectedNamespaces returns the list of filtered namespaces according to the policy namespace selector
-func GetSelectedNamespaces(included, excluded, allNamespaces []string) []string {
-	// then get the list of included
-	includedNamespaces := []string{}
+var log = ctrl.Log
+
+// Parse *Target.MatchLabels and *Target.MatchExpressions into metav1.LabelSelector for the k8s API
+func parseToLabelSelector(selector policyv1.Target) metav1.LabelSelector {
+	// Build LabelSelector from provided MatchLabels and MatchExpressions
+	var labelSelector metav1.LabelSelector
+
+	// Handle when MatchLabels/MatchExpressions were not provided to prevent nil pointer dereference.
+	// This is needed so that `include` can function independently. Not fetching any objects is the
+	// responsibility of the calling function for when MatchLabels/MatchExpressions are both nil.
+	matchLabels := map[string]string{}
+	matchExpressions := []metav1.LabelSelectorRequirement{}
+
+	if selector.MatchLabels != nil {
+		matchLabels = *selector.MatchLabels
+	}
+
+	if selector.MatchExpressions != nil {
+		matchExpressions = *selector.MatchExpressions
+	}
+
+	labelSelector = metav1.LabelSelector{
+		MatchLabels:      matchLabels,
+		MatchExpressions: matchExpressions,
+	}
+
+	return labelSelector
+}
```

Comment:
`parseToLabelSelector()` is also new to make the code reusable if `policyv1.Target` is used in the future.